src/components/ThumbnailsPanel/ThumbnailsPanel.js

```diff
@@ -55,6 +71,75 @@ class ThumbnailsPanel extends React.PureComponent {
     });
   }
 
+  onDragEnd = () => {
+    const { currentPage } = this.props;
+    const { draggingOverPageIndex, isDraggingOverTopHalf } = this.state;
+    if (draggingOverPageIndex !== null) {
+      const targetPageNumber = isDraggingOverTopHalf ? draggingOverPageIndex + 1 : draggingOverPageIndex + 2;
+
+      if (currentPage < targetPageNumber) {
+        // moving page down so target page number will decrease
+        this.afterMovePageNumber = targetPageNumber - 1;
+      } else {
+        // moving page up
+        this.afterMovePageNumber = targetPageNumber;
+      }
+
+      core.movePages([currentPage], targetPageNumber);
+    }
+
+    this.setState({ draggingOverPageIndex: null });
+  }
+
+  onPageComplete = () => {
+    if (this.afterMovePageNumber) {
```

Comment:
Hmm this one isn't called anywhere in the component. Just to confirm we need this in the later commits?

Reply:
Thank you for pointing this out, I forgot to add
```
    core.addEventListener('pageComplete', this.onPageComplete);
```

This is so if you are reordering pages, it'll go to the last page you were on. So if you are on page 1 and move it to page 5,  it'll go to page 5 (previously page 1) after it moved.